src/neptune/new/internal/disk_queue.py

```diff
@@ -39,9 +41,16 @@
 _logger = logging.getLogger(__name__)
 
 
-class DiskQueue(Generic[T]):
+@dataclass
+class QueueElement(Generic[T]):
```

Comment:
Could be declared as innser class in `DiskQueue` with an underscore prefix (it private implementation detail).
@Raalsky @shnela what do you think? Do you have some reccommendation here? Some good practices about inner classes in python?

Alternatively we can keep it exposed and return it from `get` and `get_batch` as well. I think such refactor would make good to our codebase (even if size is not used outside).

Reply:
To my knowledge inner classes are not a good pattern in general. If it's only about exposing vs non-exposing something let's indicate that with `__all__` :wink:

Reply:
After searching for _dataclass_ in out repo, I haven't found examples of an inner class, so I decided to keep QueueElement exposed. However I'm not sure myself if we should make the get() function return a QueueElement or keep it unchanged.

Reply:
I think it is always better to return dataclass over Tuple.

Reply:
If so do you think it would make sense to create a dataclass for the JsonFileSplitter as well? Currently I refactored it so it returns a tuple with a dict and its size in serialised form.

Reply:
Done